src/main/java/BinaryTree.java

```diff
@@ -0,0 +1,83 @@
+import java.util.Iterator;
+
+public interface BinaryTree<T> extends Iterable<T> {
+
+    /**
+     * Returns a reference to the element contained in the root of the tree.
+     *
+     * @return Reference to the root's element
+     */
+    T getRootElement();
+
+    /**
+     * Returns true if element is contained within the tree, false otherwise.
+     *
+     * @param element Element to search for
+     * @return True if element is in tree, false otherwise
+     */
+    boolean contains(T element);
+
+    /**
+     * Returns a reference to the element searched for within the tree. If
+     * the element is not found, this method throws an exception.
+     *
+     * @param element element to search the tree for
+     * @return Reference to the target element
+     */
+    T find(T element);
+
+    /**
+     * Checks if the tree is empty.
+     *
+     * @return True if the tree is empty, false otherwise.
+     */
+    boolean isEmpty();
+
+    /**
+     * Returns the number of elements in the tree.
+     *
+     * @return Count of the number of elements in the tree
+     */
+    int size();
+
+    /**
+     * Iterator over all elements in the binary tree.
+     *
+     * @return Iterator for the binary tree
+     */
+    Iterator<T> iterator();
+
+    /**
+     * Iterator over all elements in the tree in a 'pre-order'
+     * ordering.
+     *
+     * @return Pre-order iterator for tree
+     */
+    Iterator<T> iteratorPreOrder();
```

Comment:
Minor, it's more typical to see specialized iterators named like `preOrderIterator`

Reply:
> Overall, not sure how much an interface is needed. I think binary search on an array can help introduce optimal sorted insertion and the linked graph implementation for the search tree structure.

The idea is twofold:
1. It can be used to reinforce inheritance
2. There _was_ going to be a general linked implementation of a binary tree

That said, I am thinking of cutting the more general one as the adds & removes may be too complex for this stage, thus adding noise that's not needed. 

For now, perhaps keep the Interface and see how things plan out.

Reply:
**OR**, did you mean both interfaces, and just make a `LinkedBinarySearchTree`?

Reply:
I mean, perhaps the topic order can be:
1. binary search in array (recursion, complexity, all that jazz)
2. binary search tree, linked implementation (array impl is more on the non-trivial side)
3. implement `LinkedSortedBag` (imo, inheritance was touched on sufficiently; plus we still have sorting to visit [TYPE CLASSES HYPE {even though java doesn't support them!}])
4. maybe re-implement search methods in `ArraySortedBag`

Reply:
These are all fantastic ideas and hit on in #202. 
I'll still make a `BinarySerachTree` interface for them to leave an array implementation on the table for maybe a lab or something.